### PR TITLE
feature/1.2-save_feature

### PR DESCRIPTION
**Pull request by imranmuyingo**
This is an implementation of a save feature used to save the game scores and the Boogle game to be played at a later time

### CODE REVIEW

File: boggle/BoogleLoad.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        selectBoardBox.setAlignment(Pos.CENTER);
+
+        dialogVbox.getChildren().add(selectBoardBox);
+        Scene dialogScene = new Scene(dialogVbox, 400, 400);
+        dialog.setScene(dialogScene);
+        dialog.show();
+//        dialog.setOnCloseRequest(event -> {
+//            tetrisView.paused = false;
+//        });
```
Comment (evanMWJeong) [inline boggle/BoogleLoad.java]: we can just delete this whole segment

File: boggle/BoggleGame.java [hunk 2]
```diff
@@ -317,4 +318,17 @@ private void computerMove(Map<String,ArrayList<Position>> all_words){
     }
     }
 
+    public void saveGame(File fileName) {
```
Comment (gideonmehna) [inline boggle/BoggleGame.java:360]: Please Document what this function does

File: boggle/BoggleSave.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    public class BoggleSave {
+
+    BoggleGame game;
+
+    private Label saveFileErrorLabel = new Label("");
+    private Label saveBoardLabel = new Label(String.format("Enter name of file to save"));
+    private TextField saveFileNameTextField = new TextField("");
+    private Button saveBoardButton = new Button("Save board");
+
+
```
Comment (gideonmehna) [inline boggle/BoggleSave.java:31]: Just documentation

**Review by gideonmehna — COMMENTED** [review]
All good, just please document the functions in boggle view.

File: views/BoggleView.java [hunk 4]
```diff
@@ -585,7 +614,24 @@ private void giveEndRoundInstructions(){
             instructionsBox.getChildren().clear();
             giveFirstInstructions();
         });
+        
 
     }
 
+    public BoggleGame getModel() {
```
Comment (gideonmehna) [inline views/BoggleView.java:752]: Add some documentation to this too. We are going to generate a java doc

**Pull request merged**